addons/stock/models/stock_move.py

```diff
@@ -1104,40 +1104,42 @@ def remove_prefix(text, prefix):
         return vals_list
 
     def _push_apply(self):
-        new_moves = []
+        all_new_moves = []
         for move in self:
+            new_moves = []
             new_move = self.env['stock.move']
-
-            # if the move is a returned move, we don't want to check push rules, as returning a returned move is the only decent way
-            # to receive goods without triggering the push rules again (which would duplicate chained operations)
-            # first priority goes to the preferred routes defined on the move itself (e.g. coming from a SO line)
             warehouse_id = move.warehouse_id or move.picking_id.picking_type_id.warehouse_id
-
             StockRule = self.env['stock.rule']
             if move.location_dest_id.company_id not in self.env.companies:
                 StockRule = self.env['stock.rule'].sudo()
-                move = move.with_context(allowed_companies=self.env.user.company_ids.ids)
+                move = move.with_context(allowed_companies=move.env.user.company_ids.ids)
                 warehouse_id = False
-
             related_packages = self.env['stock.package'].search_fetch([('id', 'parent_of', move.move_line_ids.result_package_id.ids)], ['package_type_id'])
 
-            rule = StockRule._get_push_rule(move.product_id, move.location_dest_id, {
-                'route_ids': move.route_ids | related_packages.package_type_id.route_ids, 'warehouse_id': warehouse_id, 'packaging_uom_id': move.packaging_uom_id,
-            })
-
-            excluded_rule_ids = []
-            while (rule and rule.push_domain and not move.filtered_domain(literal_eval(rule.push_domain))):
-                excluded_rule_ids.append(rule.id)
-                rule = StockRule._get_push_rule(move.product_id, move.location_dest_id, {
-                    'route_ids': move.route_ids | related_packages.package_type_id.route_ids, 'warehouse_id': warehouse_id, 'packaging_uom_id': move.packaging_uom_id,
-                    'domain': [('id', 'not in', excluded_rule_ids)],
-                })
-
-            # Make sure it is not returning the return
-            if rule and (not move.origin_returned_move_id or move.origin_returned_move_id.location_dest_id.id != rule.location_dest_id.id):
-                new_move = rule._run_push(move) or new_move
-                if new_move:
+            if move.move_line_ids:
```

Comment:
remove the if and the else section. We only apply push on done and to pass a move to done state, it needs sml